cpp/src/arrow/util/parsing.h

```diff
@@ -351,6 +358,121 @@ class StringConverter<Int32Type> : public StringToSignedIntConverterMixin<Int32T
 template <>
 class StringConverter<Int64Type> : public StringToSignedIntConverterMixin<Int64Type> {};
 
+template <>
+class StringConverter<TimestampType> {
```

Comment:
Do you have a sense of performance of this?

Reply:
No idea. Hopefully it shouldn't be too slow...